### PR TITLE
SPRIND-14

### PR DESCRIPTION
**Pull request by sksadjad**
added x5c support and made sure that we support request-responses without dids

### CODE REVIEW

File: packages/client/lib/__tests__/IT.spec.ts [hunk 1]
```diff
@@ -222,164 +287,36 @@ describe('OID4VCI-Client should', () => {
...
-        proofInput: proof,
-        credentialTypes: credentialOffer.original_credential_offer.credential_configuration_ids[0],
-      });
-      expect(credResponse.successBody?.credential).toEqual(mockedVC);
-    },
-    UNIT_TEST_TIMEOUT,
-  );
-});
-
-describe('OIDVCI-Client for v1_0_13 should', () => {
```
Comment (nklomp) [inline packages/client/lib/__tests__/IT.spec.ts:292]: Why is this removed?

File: packages/client/lib/__tests__/IT.spec.ts [hunk 2]
```diff
@@ -222,164 +287,36 @@ describe('OID4VCI-Client should', () => {
...
-    nock(ISSUER_URL)
-    .post(/credential/)
-    .reply(200, {
-      format: 'jwt-vc',
-      credential: mockedVC,
-    });
-  }*/
-
-  it('should successfully resolve the credential offer URI', async () => {
-    const uri = 'https://mijnkvk.acc.credenco.com/openid4vc/credentialOffer?id=32fc4ebf-9e31-4149-9877-e3c0b602d559';
```
Comment (nklomp) [inline packages/client/lib/__tests__/IT.spec.ts:339]: Why is this removed?

**Review by nklomp — CHANGES_REQUESTED** [review]
Some questions on removed tests

File: packages/client/lib/__tests__/IT.spec.ts (hunk 1 above)
Comment (sksadjad) [reply packages/client/lib/__tests__/IT.spec.ts:292]: added back. not sure why it's not showing it here

File: packages/client/lib/__tests__/IT.spec.ts (hunk 2 above)
Comment (sksadjad) [reply packages/client/lib/__tests__/IT.spec.ts]: added back

File: packages/client/lib/CredentialRequestClientBuilder.ts [hunk 3]
```diff
@@ -40,24 +23,31 @@ export class CredentialRequestClientBuilder {
...
+      return CredentialRequestClientBuilderV1_0_13.fromCredentialIssuer({
+        credentialIssuer,
+        metadata,
+        version,
+        credentialIdentifier,
+        credentialTypes,
+      });
+    } else {
+      if (!credentialTypes || credentialTypes.length === 0) {
+        throw new Error('CredentialTypes must be provided for v1_0_11');
```
Comment (nklomp) [inline packages/client/lib/CredentialRequestClientBuilder.ts:58]: Why doesn't the v11 builder take care of this?

**Review by nklomp — COMMENTED** [review]
See remark about code outside of the v11 builder doing a check, that probably the v11 builder should do itself

File: packages/client/lib/CredentialRequestClientBuilder.ts (hunk 3 above)
Comment (sksadjad) [reply packages/client/lib/CredentialRequestClientBuilder.ts:58]: this is because in the `redentialRequestClientBuilderV1_0_11.fromCredentialIssuer` the `credentialTypes` param is required and in the draft13 it is optional. so I think it's better for it to be optional in the general `fromCredentialIssuer` function and we only check if it's passing a value or not.
